src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -86,21 +86,29 @@ public List<RawEntityTypeSummary> getEntityTypeSummary(Set<UUID> entityTypeIds)
       .map(row -> new RawEntityTypeSummary(row.get(idField), row.get(nameField)));
   }
 
-  public void replaceEntityTypeDefinitions(List<EntityType> entityTypes) throws JsonProcessingException {
+  public void replaceEntityTypeDefinitions(List<EntityType> entityTypes) {
     log.info("Replacing entity type definitions with new set of {} entities", entityTypes.size());
 
     // we use this instead of truncate since it has better rollback support
-    jooqContext.deleteFrom(table(TABLE_NAME)).execute();
-
-    InsertValuesStep2<Record, UUID, JSONB> insert = jooqContext
-      .insertInto(table(TABLE_NAME))
-      .columns(field(ID_FIELD_NAME, UUID.class), field(DEFINITION_FIELD_NAME, JSONB.class));
-
-    for (EntityType entityType : entityTypes) {
-      insert.values(UUID.fromString(entityType.getId()), JSONB.jsonb(objectMapper.writeValueAsString(entityType)));
-    }
-
-    insert.execute();
+    jooqContext.transaction(transaction -> {
```

Comment:
This transaction made this diff much uglier than it seems. The only changes are the introduction of the transaction and the change to how this does its `deleteFrom()`